### ISSUE TITLE
Support openshift oc client when applying resources (i.e. routes)

### ISSUE THREAD

**Issue by peterbosalliandercom** [argoproj/argo-cd#1734]
**Is your feature request related to a problem? Please describe.**
When deploying a helm stable chart (for example sonatype-nexus) we like to create a route for openshift. This is supported by helm stable chart: https://raw.githubusercontent.com/helm/charts/master/stable/sonatype-nexus/templates/route.yaml

After deploying it with argocd an error occurs because argocd runs kubectl cli and not the openshift OC cli. When deploying this: 

```
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  name: nexus
  labels:
  annotations:
spec:
  host: xxxxxxxxxxxxxxxx
  port:
    targetPort: nexus-service
  tls:
    insecureEdgeTerminationPolicy: Redirect
    termination: edge
  to:
    kind: Service
    name: sonatype-nexus-service
    weight: 100
  wildcardPolicy: None
```

The error is:
error: error validating "route.yaml": error validating data: ValidationError(Route): missing required field "status" in com.github.openshift.api.route.v1.Route; if you choose to ignore these errors, turn validation off with --validate=false

**Describe the solution you'd like**
Not sure about this but it would be nice that argocd also is able to use the oc client during syncing because when running oc create -f route.yaml works fine with above yaml file.
Don't know if this is a good solution. 

**Have you thought about contributing yourself?**

Open Source software thrives with your contribution. It not only gives skills you might not be able to get in your day job, it also looks amazing on your resume. 

If you want to get involved, check out the 
[contributing guide](https://github.com/argoproj/argo-cd/blob/master/docs/CONTRIBUTING.md), then reach out to us on [Slack](https://argoproj.github.io/community/join-slack) so we can see how to get you started.

**Comment by alexec** [comment]
Hi - can you please provide exact repo steps?

**Comment by peterbosalliandercom** [comment]
Repo steps:
1.  sync the following Route in argocd on openshift cluster:
```
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  name: nexus
  labels:
  annotations:
spec:
  host: xxxxxxxxxxxxxxxx
  port:
    targetPort: nexus-service
  tls:
    insecureEdgeTerminationPolicy: Redirect
    termination: edge
  to:
    kind: Service
    name: sonatype-nexus-service
    weight: 100
  wildcardPolicy: None
```
2. Error occurs:
error: error validating "route.yaml": error validating data: ValidationError(Route): missing required field "status" in com.github.openshift.api.route.v1.Route; if you choose to ignore these errors, turn validation off with --validate=false

**Comment by alexec** [comment]
Ok. I don't currently have access to OpenShift to test, but I think this would be solved #1063

**Comment by alexmt** [comment]
As a workaround, you can include empty status into Route manifest and use custom diffing to ignore differences in route status:

argocd-cm ConfigMap:
```
  resource.customizations: |
    route.openshift.io/Route:
      ignoreDifferences: |
        jsonPointers:
        - /status
```

**Comment by alexec** [comment]
Should be fixed in v1.1

**Comment by peterbosalliandercom** [comment]
Thanks Alexec!
